book_review_system/src/main/java/app/service/impl/BackgroundJob.java

```diff
@@ -0,0 +1,40 @@
+package app.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.mail.MailSender;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import app.service.RequestService;
+import app.util.MailUtil;
+
+@Configuration
+@EnableScheduling
+@PropertySource("classpath:datasources.properties")
+public class BackgroundJob {
+	@Autowired
+	private MailSender mailSender;
+	@Autowired
+	private RequestService requestService;
+	@Value("${mail.admin}")
+	private String adminMail;
+	@Value("${mail.username}")
+	private String mailName;
+	
+	// Send Request 12am EveryDay
+	@Scheduled(cron = "0 0 12 * * ?")
+	public void sendListRequest() {
+		try {
+			SimpleMailMessage message = MailUtil.message(mailName, adminMail, "Request Handle",
```

Comment:
`"Request Handle"` `"Request not Handle: "` dùng message.properties đi em

Reply:
Cái Scheduled nó k cho truyền tham số vào method chị ơi, nên đa ngôn ngữ được chị ạ.

Reply:
https://www.petrikainulainen.net/programming/spring-framework/spring-from-the-trenches-using-environment-specific-cron-expressions-with-the-scheduled-annotation/
Cái này `"0 0 12 * * ?"` cũng khai báo vào properties đi em